### PR TITLE
GPS-13-Der-skal-laves-pagination-p-produkter

### PR DESCRIPTION
**Pull request by Gi5lund**

### CODE REVIEW

File: src/main/java/dk/kea/project/api/StoreController.java [hunk 1]
```diff
@@ -1,29 +1,136 @@
...
+import java.time.LocalDateTime;
+import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @RequestMapping("/api/stores")
 public class StoreController {
-    SallingService sallingService;
-    public StoreController(SallingService sallingService) {
+    @Autowired
```
Comment (steffengroenandersen) [inline src/main/java/dk/kea/project/api/StoreController.java:28]: Consider doing constructor dependency injection and remove @Autowired

File: src/main/java/dk/kea/project/api/StoreController.java [hunk 2]
```diff
@@ -1,29 +1,136 @@
...
+        });
+        storeService.addStores(stores);
+
+        return sallingService.getAllStores();
+    }
 
     @GetMapping("/foodwaste")
-    public List<ProductResponse> getFoodWaste(@RequestParam String id){
+    public List<SallingResponse> getFoodWaste(@RequestParam String id){
+        // kald salling service brug List<SallingResponse> i ny service
```
Comment (steffengroenandersen) [inline src/main/java/dk/kea/project/api/StoreController.java:65]: Consider having comments in english.

File: src/main/java/dk/kea/project/api/StoreController.java [hunk 3]
```diff
@@ -1,29 +1,136 @@
...
+            int requestId = productService.findNewestRequest(id,now.minusMinutes(15)).getRequestId();
+           List<SallingResponse> sallingResponse = sallingService.getFoodWaste(id);
+         convertSallingResponse(sallingResponse, requestId, id);
+//            productService.addProducts(products);
+
+
+        }
+        return productService.getProducts(id);
+    }
+//    public List<ProductResponse> getProducts(@RequestParam String id){
```
Comment (steffengroenandersen) [inline src/main/java/dk/kea/project/api/StoreController.java:90]: Consider removing out-commented code.

File: src/main/java/dk/kea/project/api/StoreController.java [hunk 4]
```diff
@@ -1,29 +1,136 @@
...
+//            int requestId = productService.findNewestRequest(id,now.minusMinutes(15)).getRequestId();
+//            List<SallingResponse> sallingResponse = sallingService.getFoodWaste(id);
+//            convertSallingResponse(sallingResponse, requestId, id);
+////            productService.addProducts(products);
+//
+//
+//        }
+//        return productService.getProducts(id);
+//    }
+    private void convertSallingResponse(List<SallingResponse> sallingResponse,int requestId, String storeId) {
```
Comment (steffengroenandersen) [inline src/main/java/dk/kea/project/api/StoreController.java:111]: Consider having this type of logic in the service class.

File: src/main/java/dk/kea/project/dto/ProductResponse.java [hunk 5]
```diff
@@ -1,42 +1,35 @@
...
+		public String description;
+		public String ean;
+		public String image;
+		public double originalPrice;
+		public double newPrice;
+		public double discount;
+		public double percentDiscount;
+		public String category;
 
-//    public String toString(){
```
Comment (steffengroenandersen) [inline src/main/java/dk/kea/project/dto/ProductResponse.java:15]: Consider removing out-commented code.

File: src/main/java/dk/kea/project/repository/StoreRepository.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package dk.kea.project.repository;
+
+import dk.kea.project.entity.Store;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface StoreRepository extends JpaRepository<Store, String>
+{
+
+	}
```
Comment (steffengroenandersen) [inline src/main/java/dk/kea/project/repository/StoreRepository.java:9]: Formatting looks weird :D

File: src/main/java/dk/kea/project/service/ProductService.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.data.domain.Pageable;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ProductService
+	{
+		@Autowired
```
Comment (steffengroenandersen) [inline src/main/java/dk/kea/project/service/ProductService.java:21]: Consider doing constructor dependency injection instead of Autowired

**Review by steffengroenandersen — APPROVED** [review]
Nice work!

Consider adding JavaDoc and tests.

File: src/main/java/dk/kea/project/api/StoreController.java (hunk 1 above)
Comment (Gi5lund) [reply src/main/java/dk/kea/project/api/StoreController.java:28]: check

**Pull request merged**